File: native/app/src/main/java/com/ubb/album_manager/domain/Album.kt
package com.ubb.album_manager.domain

import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "albums")
data class Album(
    val name: String = "",
    val artist: String = "",
    val releaseYear: Int = 2000,
    val genre: String = "",
    val url: String = DEFAULT_URL,
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0
) {
    companion object {
        const val DEFAULT_URL = "https://media.istockphoto.com/id/" +
                "92889831/photo/cd-dvd-top.jpg?s=612x612&w=0&k=20" +
                "&c=OMRPZLp3frbSFyoi_FRsLiSay3XM12aTF-fraidqcxE="

        const val INVALID_ID = -1

        val DEFAULT_ALBUM = Album(
            name = "Unknown",
            artist = "Unknown Artist",
            releaseYear = 0,
            genre = "Unknown Genre",
            url = "",
            id = INVALID_ID
        )
    }

    override fun toString(): String {
        return """album: {
                    id: $id
                    name: $name
                    artist: $artist
                    genre: $genre
                    release_year: $releaseYear
                    url: $url
                  }"""
    }
}

File: non-native/lib/model/album.dart
import 'package:uuid/uuid.dart';

const String defaultUrl = "";
const String defaultId = "00000000-0000-0000-0000-000000000000";
class Album {
  final String name;
  final String artist;
  final int releaseYear;
  final String genre;
  final String url;
  final String id;

  Album({
    this.name = "",
    this.artist = "",
    this.releaseYear = 0,
    this.genre = "",
    this.url = defaultUrl,
    String? id,
  }) : id = id ?? const Uuid().v4();

  @override
  String toString() {
    return '''Album: {
                id: $id,
                name: $name,
                artist: $artist,
                genre: $genre,
                release_year: $releaseYear,
                url: $url
              }''';
  }
  Album copyWith({
    String? name,
    String? artist,
    int? releaseYear,
    String? genre,
    String? url,
    String? id,
  }) {
    return Album(
      name: name ?? this.name,
      artist: artist ?? this.artist,
      releaseYear: releaseYear ?? this.releaseYear,
      genre: genre ?? this.genre,
      url: url ?? this.url,
      id: id ?? this.id,
    );
  }
}
